### PR TITLE
OSX support

### PR DESCRIPTION
**Pull request by notyourav**
Tested on WSL and PowerShell. Had to add a step for creating the build directories, which is the only thing that might break.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,13 +42,20 @@ else
   WINE := wine
 endif
 
+# Hack for OSX
+ifeq ($(UNAME_S),Darwin)
+	CPP     := cpp-10 -P
```
Comment (barzamin) [inline Makefile:47]: is `cpp-10` tied to a specific macos version

File: Makefile [hunk 2]
```diff
@@ -79,12 +87,13 @@ SBSS_PDHR := 10
...
 default: all
 
-all: $(DOL)
+all: dirs $(DOL)
 
 ALL_DIRS := build $(BUILD_DIR) $(addprefix $(BUILD_DIR)/,$(SRC_DIRS) $(ASM_DIRS))
 
 # Make sure build directory exists before compiling anything
-DUMMY != mkdir -p $(ALL_DIRS)
+dirs:
```
Comment (barzamin) [inline Makefile:95]: add `dirs` to `.PHONY`

**Review by barzamin — APPROVED** [review]
🍎

File: Makefile (hunk 1 above)
Comment (notyourav) [reply Makefile:47]: No, but it is the only alternative I could come up with. Apple decided to claim `gcc`, `g++`, `cpp` and others as aliases to their modified clang.

**Review by Pheenoh — APPROVED** [review]

**Pull request merged**